src/libs/actions/Task.js

```diff
@@ -151,10 +151,120 @@ function createTaskAndNavigate(currentUserEmail, parentReportID, title, descript
     Navigation.navigate(ROUTES.getReportRoute(optimisticTaskReport.reportID));
 }
 
+/**
+ * @function editTask
+ * @param {object} report
+ * @param {string} ownerEmail
+ * @param {string} title
+ * @param {string} description
+ * @param {string} assignee
+ * @returns {object} action
+ *
+ */
+
+function editTaskAndNavigate(report, ownerEmail, title, description, assignee) {
+    // Create the EditedReportAction on the task
+    const editTaskReportAction = ReportUtils.buildOptimisticEditedTaskReportAction(ownerEmail);
+
+    // If we make a change to the assignee, we want to add a comment to the assignee's chat
+    let optimisticAssigneeAddComment;
+    let assigneeChatReportID;
+    if (assignee && assignee !== report.assignee) {
+        assigneeChatReportID = ReportUtils.getChatByParticipants([assignee]);
```

Comment:
This method is returning the report not the reportID and so that is why some of the usages are still [object Object]